### DISCUSSION TITLE
Fieldcollections return no data

### DISCUSSION THREAD

**Discussion by AndreasWWylach** [pimcore/pimcore#10973] [Q&A / Support]
I been banging my head against the wall on this issue and meanwhile I am out of ideas.

Having an object with a localized fieldcollection containing various types ( btw, very similar to the one used in the News object - field content - on the demo site ). I extended this collection with an additional type. Nothing more and nothing less. In my development environment, I see no issues after the modification to the fieldcollection. I can use it in existing objects or view it, shows as expected. All fieldcollection values display in editmode or in the frontend in the order they were entered.
After a few more tests I transferred to code to the staging site, did the obvious like rebuild-classes, clear the cache. However, about 90% of the object field collection data returns NULL, like it was empty, all other object fields value remain intact. Event though, after a brief dev/staging comparison of the data of the field collection, the data in their `object_collection_*_localized_no` tables are still there.  But method `getPreData` in `DataObject/ClassDefinition/Data/Fieldcollections.php`, there with the call of `getObjectVar`, returns null.

I have no clue how to resolve that issue. Both used Pimcore Versions are the same, the latest V10.2.5. Like mentioned above, rebuilt classes, I cleared caches, I even removed the var/cache/ folder contents. Nothing helps.
 
I would not continue looking to resolve this issue if the data in the various fieldcollection tables wasn't there, but its seemingly all there and present.

Does anyone have an idea what I am missing here? Perhaps I been sitting to long in front of that problem and the solution is quite simple.

Thanks for any input, its greatly appreciated!

**Comment by fashxp** [comment]
Hi, did you also clear the pimcore cache? `bin/console pimcore:cache:clear`

BR
Christian

**Reply by AndreasWWylach** [reply]
> 
What's good, Christian. Yes, I did ... I ( always ) use something like `./bin/console cache:clear --no-warmup && ./bin/console pimcore:cache:clear`.

**Reply by fashxp** [reply]
then I have no idea :(

**Comment by AndreasWWylach** [comment]
OK, I got it fixed. Prevented to manually re-enter all 100+ whatsoever records :-)

Whoever is interested in the origin and solution:

The issue showed up in `DataObject/Fieldcollection/Dao.php` when I debugged the SQL-queries of the fieldcollection tables around line 61, esp. the table names and their case-sensitivity ...
Mysql table names are file system entries, so they are case insensitive if the underlying file system is. How case-sensitivity of the table names is handled, can be adjusted.
The staging machine has `lower_case_table_names` set to `0`, whereas my development environments running mariaDB has this value set to `1`. This is recommended when using InnoDB tables, thus the case with Pimcore.

So my issue was simple: When I updated the staging machine with the new object after my changes a day ago, the fieldcollection tables where created case-sensitive and messed up the query results and fieldcollections data, returning NULL at the end.

I have no control over that staging server so I fixed it myself within the DB, btw, the first time ever I hacked the Pimcore tables directly. However, such a minor change, knowing what I am doing there, is legit in such rare cases. Case closed!

**Reply by fashxp** [reply]
👍 thx for sharing!
